node/src/index.ts

```diff
@@ -33,13 +33,88 @@ export { OpenAIEmbeddingFunction } from './embedding/openai'
  */
 export async function connect (uri: string): Promise<Connection> {
   const db = await databaseNew(uri)
-  return new Connection(db, uri)
+  return new LocalConnection(db, uri)
+}
+
+export interface Connection {
+  uri: string
+
+  tableNames: () => Promise<string[]>
+
+  /**
+   * Open a table in the database.
+   *
+   * @param name The name of the table.
+   */
+  openTable: ((name: string) => Promise<Table>) & (<T>(name: string, embeddings: EmbeddingFunction<T>) => Promise<Table<T>>) & (<T>(name: string, embeddings?: EmbeddingFunction<T>) => Promise<Table<T>>)
+
+  /**
+   * Creates a new Table and initialize it with new data.
+   *
+   * @param name The name of the table.
+   * @param data Non-empty Array of Records to be inserted into the Table
+   */
+
+  createTable: ((name: string, data: Array<Record<string, unknown>>) => Promise<Table>) & (<T>(name: string, data: Array<Record<string, unknown>>, embeddings: EmbeddingFunction<T>) => Promise<Table<T>>) & (<T>(name: string, data: Array<Record<string, unknown>>, embeddings?: EmbeddingFunction<T>) => Promise<Table<T>>)
+
+  createTableArrow: (name: string, table: ArrowTable) => Promise<Table>
+
+  /**
+   * Drop an existing table.
+   * @param name The name of the table to drop.
+   */
+  dropTable: (name: string) => Promise<void>
+}
+
+export interface Table<T = number[]> {
```

Comment:
Docstring to interface.

Actually, could you take a look of whether the generated API doc looks good ?